subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6576,7 +6576,7 @@ static void ext_adv_info_fill(uint8_t evt_type, uint8_t phy, uint8_t sec_phy,
 	sep->num_reports = 1U;
 	adv_info = (void *)(((uint8_t *)sep) + sizeof(*sep));
 
-	adv_info->evt_type = evt_type;
+	adv_info->evt_type = sys_cpu_to_le16((uint16_t)evt_type);
```

Comment:
I would suggest making `evt_type` a `uint16_t` value instead of a `uint8_t` both here and in `evt_type_lookup` to avoid the casts. It does require slightly more stack, but IMO worth it to avoid any future issues with conversion

Reply:
@Tronil Are the explicit casts really needed here? Extending `uint8_t` into `uint16_t` is a safe operation, so the compiler should not complain.

Reply:
Looking at the `__bswap_16` macro which `sys_cpu_to_le16` maps to, it does a straight shift down on the input:
`#define __bswap_16(x) ((uint16_t) ((((x) >> 8) & 0xff) | (((x) & 0xff) << 8)))`
Shifting down 8 bits on a uint8_t is undefined behavior according to the C standard, so we should make sure not to do that. In practice I expect it will work as expected, but let's be safe :-)

Reply:
@Tronil as per my comment (https://github.com/zephyrproject-rtos/zephyr/pull/59367#discussion_r1234021386) you can avoid any issues by using the "right" data type to store the `evt_type` in :)

Reply:
> @Tronil as per my comment ([#59367 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/59367#discussion_r1234021386)) you can avoid any issues by using the "right" data type to store the `evt_type` in :)
True - however, `evt_type` in the legacy advertising reports is only 8 bits, and even with the extended advertising reports only the lower 8 bits are actually used (so far at least). So it is a bit a matter of opinion about what data type is "right"...
Anyway, feel free to change it if you want :-)